test/basic.html

```diff
@@ -55,22 +60,24 @@
           assert.equal(collapse.horizontal, false);
         });
 
-        test('default opened height', function(done) {
-          setTimeout(function() {
-            // store height
-            collapseHeight = getComputedStyle(collapse).height;
-            // verify height not 0px
-            assert.notEqual(collapseHeight, '0px');
-            done();
-          }, delay);
+        test('default opened height', function() {
+          // verify height not 0px
+          assert.notEqual(collapseHeight, 0);
+        });
+
+        test('opened height gets updated when inner dom changes', function() {
+          var lorem = collapse.children[1];
+          collapse.removeChild(lorem);
+          assert.isBelow(collapse.offsetHeight, collapseHeight);
+          collapse.appendChild(lorem);
+          assert.equal(collapse.offsetHeight, collapseHeight);
         });
 
         test('set opened to false', function(done) {
           collapse.opened = false;
           setTimeout(function() {
-            var h = getComputedStyle(collapse).height;
             // verify height is 0px
-            assert.equal(h, '0px');
+            assert.equal(collapse.offsetHeight, 0);
```

Comment:
I did some tests and offsetHeight considers height + padding in chrome (also it rounds to the nearest integer) http://jsbin.com/lofaxuqete
The doc though doesn't specify if it's supported in different browsers https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetHeight (also, it says it should be height + border)
In any case, iron-collapse doesn't have padding or borders, so in the tests offsetHeight and computedStyle.height should be the same.